### PR TITLE
refactor: add transifex support to user-facing messages

### PR DESCRIPTION
**Pull request by muselesscreator**
**TL;DR -**
All user-facing messages are now wrapped by react-intl tooling (through frontend-platform) to support internationalization.
**Note** this is not hooked up to actually DO the translation yet, as that is a process external to this repo

JIRA: [AU-28](https://openedx.atlassian.net/browse/AU-28)

**Developer Checklist**
- [x] Test suites passing
- [x] Documentation and test plan updated, if applicable
- [x] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Testing Instructions**

Alas, this is another one of those "I changed something fundamental, kind of everywhere" tickets where the testing is a full smoke test of making sure all of the user-facing strings still show up (and not as [Object object]).
As an example, earlier versions of this broke the dropdown default-option in cohort/track selects as mentioned above.

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [x] I've done a visual code review
- [x] I've tested the new functionality

FYI: @edx/masters-devs-gta

### CODE REVIEW

**Comment by nsprenkle** [comment]
Please add testing instructions and "what changed" if applicable.

**Comment by jansenk** [comment]
Could you break this up into multiple PRs? I know the changes are small but 95 files is a lot of files.

> Note this is not hooked up to actually DO the translation yet, as that is a process external to this repo

Do we have a ticket for this? Could you link it?

**Comment by muselesscreator** [comment]
@jansenk  the ticket is already linked

File: src/components/GradesTab/EditModal/messages.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    description: 'Edit Modal close button text',
+  },
+  visibility: {
+    id: 'gradebook.GradesTab.EditModal.contactSupport',
+    defaultMessage: 'Showing most recent actions (max 5). To see more, please contact support',
+    description: 'Edit Modal visibility hint message',
+  },
+  saveVisibility: {
+    id: 'gradebook.GradesTab.EditModal.saveVisibility',
+    defaultMessage: 'Note: Once you save your changes will be visible to students.',
```
Comment (nsprenkle) [inline src/components/GradesTab/EditModal/messages.js:41]: Comma removed from original message
```suggestion
    defaultMessage: 'Note: Once you save, your changes will be visible to students.',
```

File: src/components/GradesTab/messages.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    description: 'Alert text for invalid minimum course grade',
+  },
+  gradebookStepHeading: {
+    id: 'gradebook.GradesTab.gradebookStepHeading',
+    defaultMessage: 'Step 2: View or Modify Individual Grades',
+    description: 'Alert text for invalid minimum course grade',
+  },
+  mastersHint: {
+    id: 'gradebook.GradesTab.mastersHint',
+    defaultMessage: 'available for learners in the Master&abos;s track only',
```
Comment (nsprenkle) [inline src/components/GradesTab/messages.js:71]: Typo
```suggestion
    defaultMessage: 'available for learners in the Master&apos;s track only',
```

File: src/components/GradesTab/EditModal/messages.js [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    description: 'Edit Modal saved changes effect hint',
+  },
+  saveGrade: {
+    id: 'gradebook.GradesTab.EditModal.saveGrade',
+    defaultMessage: 'Save Gradse',
```
Comment (nsprenkle) [inline src/components/GradesTab/EditModal/messages.js:46]: Typo
```suggestion
    defaultMessage: 'Save Grades',
```

**Review by nsprenkle — CHANGES_REQUESTED** [review]
Overall looks good but I found a few typos.

Also questions about temp files & config.

File: Makefile [hunk 4]
```diff
@@ -2,8 +2,64 @@ npm-install-%: ## install specified % npm package
 	npm install $* --save-dev
 	git add package.json
 
+transifex_resource = frontend-app-gradebook
+transifex_langs = "ar,fr,es_419,zh_CN"
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply Makefile:6]: This is recommended as the default from our internal docs on setting up i18n support

File: temp/babel-plugin-react-intl/src/data/constants/filters.messages.json [hunk 5]
```diff
@@ -0,0 +1 @@
+[{"id":"gradebook.GradesTab.FilterBadges.assignment","description":"Assignment FilterBadge label","defaultMessage":"Assignment"},{"id":"gradebook.GradesTab.FilterBadges.assignmentGrade","description":"Assignment Grade FilterBadge label","defaultMessage":"Assignment Grade"},{"id":"gradebook.GradesTab.FilterBadges.assignmentType","description":"Assignment Type FilterBadge label","defaultMessage":"Assignment Type"},{"id":"gradebook.GradesTab.FilterBadges.cohort","description":"Cohort FilterBadge label","defaultMessage":"Cohort"},{"id":"gradebook.GradesTab.FilterBadges.courseGrade","description":"Course Grade FilterBadge label","defaultMessage":"Course Grade"},{"id":"gradebook.GradesTab.FilterBadges.includeCourseRoleMembers","description":"Include Course Team Members FilterBadge label","defaultMessage":"Include Course Team Members"},{"id":"gradebook.GradesTab.FilterBadges.track","description":"Track FilterBadge label","defaultMessage":"Track"}]
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply temp/babel-plugin-react-intl/src/data/constants/filters.messages.json:1]: yes!

File: src/components/BulkManagementTab/FileUploadForm.test.jsx [hunk 6]
```diff
@@ -1,23 +1,33 @@
...
-import * as appConstants from 'data/constants/app';
-
 import { FileUploadForm, mapStateToProps, mapDispatchToProps } from './FileUploadForm';
 
-const {
-  messages: { BulkManagementTab: messages },
-} = appConstants;
+import messages from './messages';
 
+/*
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply src/components/BulkManagementTab/FileUploadForm.test.jsx:19]: danke! stale.

File: src/components/GradesTab/EditModal/messages.js (hunk 1 above)
Comment (muselesscreator) [reply src/components/GradesTab/EditModal/messages.js:41]: nooooooo

File: Makefile [hunk 7]
```diff
@@ -2,8 +2,64 @@ npm-install-%: ## install specified % npm package
...
+# Pulls translations from Transifex.
+pull_translations:
+	tx pull -f --mode reviewed --language=$(transifex_langs)
+
+# This target is used by Travis.
 validate-no-uncommitted-package-lock-changes:
+	# Checking for package-lock.json changes...
 	git diff --exit-code package-lock.json
-
-test:
```
[the remark this answers is not in the document]
Comment (muselesscreator) [reply Makefile:8]: not removed.  is consumed into the above syntax (line 20)

[the remark this answers is not in the document]
Comment (nsprenkle) [reply Makefile:8]: Oop, my bad

File: src/components/GradesTab/messages.js [hunk 8]
```diff
@@ -0,0 +1,76 @@
...
+    description: 'Alert text for invalid minimum course grade',
+  },
+  gradebookStepHeading: {
+    id: 'gradebook.GradesTab.gradebookStepHeading',
+    defaultMessage: 'Step 2: View or Modify Individual Grades',
+    description: 'Alert text for invalid minimum course grade',
+  },
+  mastersHint: {
+    id: 'gradebook.GradesTab.mastersHint',
+    defaultMessage: 'available for learners in the Master&apos;s track only',
```
Comment (nsprenkle) [inline src/components/GradesTab/messages.js:71]: This isn't encoding correctly:

Screen Shot 2021-07-21 at 12 37 32

Comment (muselesscreator) [reply src/components/GradesTab/messages.js:71]: noted and fixed

**Review by nsprenkle — APPROVED** [review]
:shipit:
